src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentDeleteMethod.cs

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Maintenance.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Maintenance;
+using Microsoft.Azure.Management.Maintenance.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Maintenance
+{
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ConfigurationAssignment", DefaultParameterSetName = "DefaultParameter", SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public partial class RemoveAzureRmConfigurationAssignment : MaintenanceAutomationBaseCmdlet
+    {
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            ExecuteClientAction(() =>
+            {
+                if (ShouldProcess(this.ConfigurationAssignmentName, VerbsCommon.Remove)
+                    && (this.Force.IsPresent ||
+                        this.ShouldContinue(Properties.Resources.ResourceRemovalConfirmation,
+                                            "Remove-AzConfigurationAssignment operation")))
+                {
+                    string resourceGroupName = this.ResourceGroupName;
+                    string providerName = this.ProviderName;
+                    string resourceParentType = this.ResourceParentType;
+                    string resourceParentName = this.ResourceParentName;
+                    string resourceType = this.ResourceType;
+                    string resourceName = this.ResourceName;
+                    string configurationAssignmentName = this.ConfigurationAssignmentName;
+
+                    if (!string.IsNullOrEmpty(resourceParentType) && !string.IsNullOrEmpty(resourceParentName))
+                    {
+                        ConfigurationAssignmentsClient.DeleteParent(resourceGroupName, providerName, resourceParentType, resourceParentName, resourceType, resourceName, configurationAssignmentName);
+                    }
+                    else
+                    {
+                        ConfigurationAssignmentsClient.Delete(resourceGroupName, providerName, resourceType, resourceName, configurationAssignmentName);
+                    }
+
+                    if (this.PassThru)
```

Comment:
Always use `.IsPresent` to tell if a switch parameters is specified
```suggestion
                    if (PassThru.IsPresent)
```